files/es/web/api/service_worker_api/using_service_workers/index.md

```diff
@@ -0,0 +1,524 @@
+
+Esto se podría deber a las siguientes razones:
+
+1. No estás ejecutando tu aplicación a través de HTTPS.
+2. La ruta a tu archivo del *service worker* no está escrita correctamente — se debe escribir en relación con el origen, no con el directorio raíz de tu aplicación. En nuestro ejemplo, el trabajador está en `https://bncb2v.csb.app/sw.js` y la raíz de la aplicación es `https://bncb2v.csb.app/`. Pero la ruta se debe escribir como `/sw.js`.
+3. Tampoco está permitido apuntar a un *service worker* de un origen diferente al de tu aplicación.
+
+![](important-notes.png)
+
+También ten en cuenta:
+
+- El *service worker* solo capturará las solicitudes de los clientes bajo el alcance del *service worker*.
+- El alcance máximo para un *service worker* es la ubicación del trabajador.
+- Si tu *service worker* está activo en un cliente al que se atiende con el encabezado `Service-Worker-Allowed`, puedes especificar una lista de alcances máximos para ese trabajador.
+- En Firefox, las APIs de *Service Worker* están ocultas y no se pueden usar cuando el usuario está en [modo de navegación privada](https://support.mozilla.org/es/kb/private-browsing-use-firefox-without-history).
+
+### Instalar y activar: llena tu caché
+
+Después de que tu *service worker* esté registrado, el navegador intentará instalar y luego activar el *service worker* para tu página/sitio.
+
+El evento `install` se activa cuando una instalación se completa con éxito. El evento `install` generalmente se usa para llenar las capacidades de almacenamiento en caché sin conexión de tu navegador con los activos que necesita para ejecutar tu aplicación sin conexión. Para hacer esto, usamos la API de almacenamiento de *Service Worker*: {{domxref("cache")}} — un objeto global en *Service Worker* que nos permite almacenar los activos entregados por las respuestas y con clave de sus solicitudes. Esta API funciona de manera similar a la memoria caché estándar del navegador, pero es específica para tu dominio. Persiste hasta que le dices que no lo haga — nuevamente, tienes el control total.
+
+Así es como nuestro *service worker* maneja el evento `install`:
+
+```js
+const addResourcesToCache = async (resources) => {
+  const cache = await caches.open("v1");
+  await cache.addAll(resources);
+};
+
+self.addEventListener("install", (event) => {
+  event.waitUntil(
+    addResourcesToCache([
+      "/",
+      "/index.html",
+      "/style.css",
+      "/app.js",
+      "/image-list.js",
+      "/star-wars-logo.jpg",
+      "/gallery/bountyHunters.jpg",
+      "/gallery/myLittleVader.jpg",
+      "/gallery/snowTroopers.jpg",
+    ])
+  );
+});
+```
+
+1. Aquí agregamos un escucha de eventos `install` al *service worker* (por lo tanto, `self`), y luego encadenamos un método {{domxref("ExtendableEvent.waitUntil()") }} al evento; esto garantiza que el *service worker* no se instale hasta que el código dentro de `waitUntil()` haya ocurrido con éxito.
```

Comment:
following the localization conventions: https://github.com/mdn/translated-content/tree/main/docs/es

```suggestion
1. Aquí agregamos un detector de eventos `install` al *service worker* (por lo tanto, `self`), y luego encadenamos un método {{domxref("ExtendableEvent.waitUntil()") }} al evento; esto garantiza que el *service worker* no se instale hasta que el código dentro de `waitUntil()` haya ocurrido con éxito.
```